iothub/device/src/Messaging/IncomingMessage.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// The data structure that represents the message that is received from IotHub.
+    /// </summary>
+    public class IncomingMessage
```

Comment:
How about `CloudMessage` for C2D? Would that not make sense for module-to-module messages?

Reply:
That might work. Module-to-module communication is through direct methods, so that shouldn't cause an issue. I'll note this as an option.